src/test/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacadeTest.java

```diff
@@ -0,0 +1,108 @@
+        String someSegueAnonymousUserId = "9284723987anonymous83924923";
+
+        HttpSession httpSession = createNiceMock(HttpSession.class);
+        expect(httpSession.getAttribute(Constants.ANONYMOUS_USER)).andReturn(null).atLeastOnce();
+        expect(httpSession.getId()).andReturn(someSegueAnonymousUserId).atLeastOnce();
+        replay(httpSession);
+
+        // --- Login as a student
+        LoginResult studentLogin = loginAs(httpSession, properties.getProperty("TEST_STUDENT_EMAIL"), properties.getProperty("TEST_STUDENT_PASSWORD"));
+
+        // --- Login as an event manager
+        LoginResult eventManagerLogin = loginAs(httpSession, properties.getProperty("TEST_EVENTMANAGER_EMAIL"), properties.getProperty("TEST_EVENTMANAGER_PASSWORD"));
+
+        // --- Create a booking as a logged in student
+        HttpServletRequest createBookingRequest = createNiceMock(HttpServletRequest.class);
+        expect(createBookingRequest.getCookies()).andReturn(new Cookie[] { studentLogin.cookie }).atLeastOnce();
+        replay(createBookingRequest);
+
+        Response createBookingResponse = eventsFacade.createBookingForMe(createBookingRequest, "b34eeb0c-7304-4c25-b83b-f28c78b5d078", null);
+        // Check that the booking was created successfully
+        assertEquals(createBookingResponse.getStatus(), Response.Status.OK.getStatusCode());
+        EventBookingDTO eventBookingDTO = null;
+        if (null != createBookingResponse.getEntity() && createBookingResponse.getEntity() instanceof EventBookingDTO) {
+            eventBookingDTO = (EventBookingDTO) createBookingResponse.getEntity();
+            // Check that the returned entity is an EventBookingDTO and the ID of the user who created the booking matches
+            assertEquals(studentLogin.user.getId(), ((EventBookingDTO) createBookingResponse.getEntity()).getUserBooked().getId());
+        }
+        assertNotNull(eventBookingDTO);
+
+        // --- Check whether we are leaking PII to event managers
+        HttpServletRequest getEventBookingsByIdRequest = createNiceMock(HttpServletRequest.class);
+        expect(getEventBookingsByIdRequest.getCookies()).andReturn(new Cookie[] { eventManagerLogin.cookie }).atLeastOnce();
+        replay(getEventBookingsByIdRequest);
+
+        Response getEventBookingsByIdResponse = eventsFacade.getEventBookingsById(getEventBookingsByIdRequest, eventBookingDTO.getBookingId().toString());
+        if (null != getEventBookingsByIdResponse.getEntity() && getEventBookingsByIdResponse.getEntity() instanceof EventBookingDTO) {
+            assertNotNull(((EventBookingDTO) getEventBookingsByIdResponse.getEntity()).getUserBooked());
+            assertEquals(((EventBookingDTO) getEventBookingsByIdResponse.getEntity()).getUserBooked().getClass(), UserSummaryWithEmailAddressDTO.class);
+        }
```

Comment:
Probably right. I got tangled up in the way I thought about tests and naming the test case functions and I ended up choosing the endpoint-first approach, where I focus on one specific endpoint in each test case. But then again this left me unsatisfied because I like test cases to test for specific behaviours or bugs. Then there's the question of minimizing repetition. Anyway, I'll leave this in here for the moment and refactor it in the next PR.